xwiki-commons-core/xwiki-commons-jakartabridge/xwiki-commons-jakartabridge-servlet/src/main/java/org/xwiki/jakartabridge/servlet/JakartaServletBridge.java

```diff
@@ -0,0 +1,1002 @@
+     * @return the javax version
+     */
+    public static SessionTrackingMode toJakarta(javax.servlet.SessionTrackingMode javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        return SessionTrackingMode.valueOf(javax.name());
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Set<javax.servlet.SessionTrackingMode> toJavax(Set<SessionTrackingMode> jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return jakarta.stream().map(JakartaServletBridge::toJavax).collect(Collectors.toSet());
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static Set<SessionTrackingMode> toJakarta(Set<javax.servlet.SessionTrackingMode> javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        return javax.stream().map(JakartaServletBridge::toJakarta).collect(Collectors.toSet());
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static javax.servlet.MultipartConfigElement toJavax(MultipartConfigElement jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return new javax.servlet.MultipartConfigElement(jakarta.getLocation(), jakarta.getMaxFileSize(),
+            jakarta.getMaxRequestSize(), jakarta.getFileSizeThreshold());
+    }
+
+    /**
+     * @param javax the javax version
+     * @return the javax version
+     */
+    public static MultipartConfigElement toJakarta(javax.servlet.MultipartConfigElement javax)
+    {
+        if (javax == null) {
+            return null;
+        }
+
+        return new MultipartConfigElement(javax.getLocation(), javax.getMaxFileSize(), javax.getMaxRequestSize(),
+            javax.getFileSizeThreshold());
+    }
+
+    //////////////////////////////////////////////////
+    // Multi
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Collection<javax.servlet.http.Part> toJavax(Collection<Part> jakarta)
+    {
+        return JakartaBridge.toJavax(jakarta, JavaxToJakartaPart::new, JakartaToJavaxPart::new);
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Collection<Part> toJakarta(Collection<javax.servlet.http.Part> jakarta)
+    {
+        return JakartaBridge.toJakarta(jakarta, JavaxToJakartaPart::new, JakartaToJavaxPart::new);
+    }
+
+    /**
+     * @param jakarta the jakarta version
+     * @return the javax version
+     */
+    public static Enumeration<javax.servlet.Servlet> toJavax(Enumeration<Servlet> jakarta)
+    {
+        if (jakarta == null) {
+            return null;
+        }
+
+        return Collections.enumeration(Collections.list(jakarta).stream().map(JakartaServletBridge::toJavax).toList());
```

Comment:
Converting first to a list seems quite a waste, according to [this answer](https://stackoverflow.com/a/33245847) it should also be possible to directly create a stream based on an iterator. Otherwise, I'm wondering if it wouldn't be cleaner to use a regular for-loop to directly create a list with the correct elements.

Reply:
I was lazy, but what would actually make the most sense in terms of performance is creating an Enumeration wrapper which manipulate the other one in its implementation, like I did for the other collections in `org.xwiki.jakartabridge.internal`. I'm going to add one more helper for Enumeration.